packages/block-editor/src/components/block-list/index.js

```diff
@@ -128,64 +145,14 @@ class BlockList extends Component {
 
 		selection.removeAllRanges();
 		selection.addRange( range );
-	}
-
-	/**
-	 * Binds event handlers to the document for tracking a pending multi-select
-	 * in response to a mousedown event occurring in a rendered block.
-	 *
-	 * @param {string} clientId Client ID of block where mousedown occurred.
-	 */
-	onSelectionStart( clientId ) {
-		if ( ! this.props.isSelectionEnabled ) {
-			return;
-		}
-
-		this.startClientId = clientId;
-		this.props.onStartMultiSelect();
-
-		// `onSelectionStart` is called after `mousedown` and `mouseleave`
-		// (from a block). The selection ends when `mouseup` happens anywhere
-		// in the window.
-		window.addEventListener( 'mouseup', this.onSelectionEnd );
-
-		// Removing the contenteditable attributes within the block editor is
-		// essential for selection to work across editable areas. The edible
-		// hosts are removed, allowing selection to be extended outside the
-		// DOM element. `onStartMultiSelect` sets a flag in the store so the
-		// rich text components are updated, but the rerender may happen very
-		// slowly, especially in Safari for the blocks that are asynchonously
-		// rendered. To ensure the browser instantly removes the selection
-		// boundaries, we remove the contenteditable attributes manually.
-		Array.from(
-			this.ref.current.querySelectorAll( '.rich-text' )
-		).forEach( ( node ) => {
-			node.removeAttribute( 'contenteditable' );
-		} );
-	}
+	} );
```

Comment:
So this runs on each render, just making sure it's expected.

Reply:
I guess we could make it depend on `multiSelectedBlockClientIds` (not currently the case in `master`). Let me check if it would work properly.

Reply:
Fixed. :)